### DISCUSSION TITLE
Auto-complete command from pyFluent console

### DISCUSSION THREAD

**Discussion by RandSim** [ansys/pyfluent#1568] [Q&A] [answered]
Hi,
Is there an auto-complete feature when using pyFluent console in fluent solver mode? Tab doesn't auto-complete for me.
Thanks,
Alex,

**Comment by mkundu1** [comment, marked as answer]
If you are using the latest pyfluent package (0.13) or pyfluent github source with generated API files from Fluent 23.1, auto-completion within the solver level will be available:

![2023-05-01 20_43_49-● test py - Visual Studio Code  Administrator](https://user-images.githubusercontent.com/94432368/235556531-b494f94f-96ec-491d-b43f-04a9012e8bcc.png)

If you mean the Python console in standalone Fluent, please check the Preferences > Appearance > Console auto-completer setting.

**Reply by RandSim** [reply]
Thanks! That worked well!

**Comment by dnwillia-work** [comment]
@RandSim There is also pyreadline3 which I sometimes use when not in VS Code (raw commandline).

https://pypi.org/project/pyreadline3/
